csrc/python_frontend/fusion_definition.cpp

```diff
@@ -170,6 +175,26 @@ std::vector<at::Tensor> FusionDefinition::execute(
 
   auto scheds = fusionCache()->queryFusionSchedules(id().value());
 
+#ifdef NVFUSER_DISTRIBUTED
```

Comment:
Doesn't seem to be necessary? In general, I'd like to reduce the use of the macro, which makes single- and multi-device build diverge.

Reply:
We are using `multi_device_executor` which is `std::unique_ptr<MultiDeviceExecutor>`. `MultiDeviceExecutor` is hidden behind the same macro `NVFUSER_DISTRIBUTED`.

single/multi-device build has to diverge, since our mult-device build relies on PyTorch package.
PyTorch could be shipped without multi-device support and we still want nvfuser to be able to built against that.

Reply:
> MultiDeviceExecutor is hidden behind the same macro NVFUSER_DISTRIBUTED.

You are right -- I missed that! Aha, I can see we'll soon check NVFUSER_DISTRIBUTED "everywhere" :) It's so easy to have to propagate it. 

I'll look more into this. It may be possible to confine the use in just the communicator -- the communicator may be able to return the same tensor when the world size is one instead of calling c10d (which is guarded behind NVFUSER_DISTRIBUTED AFAIK). 

> single/multi-device build has to diverge

Yeah, my point was to reduce the build divergence because more build divergence means more likely to see build-succeeds-on-my-machine-but-fails-on-CI.

Reply:
> You are right -- I missed that! Aha, I can see we'll soon check NVFUSER_DISTRIBUTED "everywhere" :) It's so easy to have to propagate it.
> 
> I'll look more into this. It may be possible to confine the use in just the communicator

You are right! It's been on our TODO list with @cowanmeg for quite some time now :) Some non-trivial refactor and interface will be needed though in communicator.cpp/.h and communication.cpp/.h
If it becomes higher priority let me know